src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -237,7 +267,55 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
       if (copiedStreamingBuffer != null) {
         insertBufferedRecords(copiedStreamingBuffer);
       }
+    } else {
+      LOGGER.debug(
+          "Skip adding offset:{} to buffer for channel:{} because"
+              + " offsetPersistedInSnowflake:{}, processedOffset:{}",
+          kafkaSinkRecord.kafkaOffset(),
+          this.getChannelName(),
+          this.offsetPersistedInSnowflake.get(),
```

Comment:
We might want to pull this out above the `if` statement and use it in the `if` conditional and in the else block as the value could change by the time we log this which could make debugging tricky.

Reply:
Actually the `offsetPersistedInSnowflake` would not change elsewhere. It will only change after it fetches the offsetToken when this `put` operation is completed and it goes into `preCommit`. It happens in the same thread and `offsetPersistedInSnowflake` is not set in separate thread.

Reply:
I see makes sense. Maybe pull it out anyway so we can reduce the number of calls to `get()`?

Reply:
Also, per row logging seems too much?

Reply:
Will put it out at top.. 
Re: per row logging-> only in case of skipped offsets + debug level log, so should be fine IMO.